### PR TITLE
gobject-introspection: update DEPENDS to python3

### PR DESCRIPTION
**Pull request by tbeckerson**
optional_depends was using python2, updated to use python3 (default)

### CODE REVIEW

**Comment by lazyape** [comment]
my feeling is that we should remove that python dependency from the DEPENDS in general
meson itself is depended on python, why having a question for supporing python without
any compilation arguments when it is already installed?
According to its meson.build, python is used if doctool options is enabled (true by default).
I would go one step further and i would disable doctool by default in moonbase, users wanting
that future can enable it by editing the module throught zlocal and enabling it there.

**Comment by tbeckerson** [comment]
lazyape;

Removing doctool - I think that's a bigger suggestion that needs discussion with the other contributors, and it's beyond the scope of this commit.

In this specific commit to gobject-introspection (GI),  your suggestion made me look at this in more detail. 
The meson.build file looks like GI has a hard dependency on python (lines 18-22).
The 'python' option shown in meson_options.txt is to declare what version of python to use, not a yes/no to using it.

I now believe the optional_depends in this module was probably included a long time ago, so a user could specifically install python2 if they wanted GI to use it instead.

tl;dr - I agree that the optional_depends should be removed entirely. I'll withdraw this PR and submit a new one with this change.
